### ISSUE TITLE
Compiler throws panic on a function w/o return value

### ISSUE THREAD

**Issue by AlexVanin** [nspcc-dev/neo-go#929]
neo-go master-2.x branch
neo-go version 0.74.1-pre-148-ge097e86b

I got a `loop` function with `for` loop over slice, stored in the contract storage. 

```go
package foo

import "github.com/nspcc-dev/neo-go/pkg/interop/storage"

func loop(ctx storage.Context, key string, value int) {
	list := storage.Get(ctx, key).([]int)
	for i := 0; i < len(list); i++ {
		if list[i] == value {
			return
		}
	}
	list = append(list, value)
	storage.Put(ctx, key, list)
}

func Main() int {
	ctx := storage.GetContext()
	loop(ctx, "foo", 1)
	return 1
}
```

When I run this code in a tests, I expect to get a storage error, something like this one.
```
error encountered at instruction 71 (SYSCALL): failed to invoke syscall: could not find [102 111 111]
```

But instead compiler throws panic.

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
```

If I add return value in a `loop` function then it compiles correctly:

```go
func loop(ctx storage.Context, key string, value int) int {
	list := storage.Get(ctx, key).([]int)
	for i := 0; i < len(list); i++ {
		if list[i] == value {
			return 0
		}
	}
	list = append(list, value)
	storage.Put(ctx, key, list)
	return 0
}
```

Are there any special rules for functions without return values?

**Comment by roman-khimov** [comment]
I'm not sure whether it's caused by attempt to Put/Get `[]int` (which probably won't work) or function return value.

**Comment by AlexVanin** [comment]
@roman-khimov You can omit storage operations and you'll have the same issue (`loop` function will be kinda useless though)

```go
func loop(ctx storage.Context, key string, value int) {
	var list = []int{2,3,4,5,6}
	for i := 0; i < len(list); i++ {
		if list[i] == value {
			return 
		}
	}
}
```

**Comment by fyrchik** [comment]
It is cause not by the function itself, but rather by `return` with no arguments.

**Comment by fyrchik** [comment]
Closed via #931 .
